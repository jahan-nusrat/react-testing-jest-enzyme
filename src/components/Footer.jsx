import React from 'react'

const Footer = () => {
  return (
    <div>
      <hr />
      <span className='phone'>Customer Service: 1-889-34343-43</span>
    </div>
  )
}

export default Footer
