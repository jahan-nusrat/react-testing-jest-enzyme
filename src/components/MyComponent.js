import React, { useState } from 'react'

const styles = {
  fontFamily: 'monospace',
  width: '500px',
  margin: 'auto',
  btn : {
    padding: '0.5rem 1rem',
    backgroundColor: 'lightgreen',
    outline: 'none',
    border: '1px solid yellow',
    fontSize: '15px',
    borderRadius: '5px',
    cursor: 'pointer'
  }
}

const MyComponent = () => {
  const [state, setState] = useState(true)
  const toggleText = () => {
    setState(!state)
  }
  return (
    <div style={styles}>
      <button style={styles.btn} onClick={toggleText}>Toggle</button>
      {state && <p>Lorem ipsum dolor sit</p>}
    </div>
  )
}

export default MyComponent
