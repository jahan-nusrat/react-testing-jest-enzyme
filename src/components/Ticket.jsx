import React, { useState } from 'react'

const Ticket = (props) => {
  const [count, setCount] = useState(0)
  const increment = () => {
    setCount((prevState) => prevState + 1)
  }
  return (
    <div style={{width: '300px', margin: 'auto', textAlign:'center'}}>
      <h2 className='title'>{props.name}</h2>
      <button style={{padding:'0.5rem 1rem', backgroundColor: 'yellow', cursor: 'pointer'}} onClick={increment}>Add</button>
      <h2 className='total'>Count: {count}</h2>
    </div>
  )
}

export default Ticket
