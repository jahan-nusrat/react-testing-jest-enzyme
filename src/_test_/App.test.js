import React from 'react'
import { mount, shallow } from 'enzyme'
import MyComponent from '../components/MyComponent'
import Footer from '../components/Footer'
import MyMainApp from '../components/Account/MyMainApp'
import Ticket from '../components/Ticket'
import toJson from 'enzyme-to-json'
import renderer from 'react-test-renderer'

const name = 'Hello world'

describe('componenets rendering without creashing', () => {
  it('MyComponent', () => {
    shallow(<MyComponent />)
  });
  it('Footer render', () => {
    shallow(<Footer />)
  });
  it('MyMainApp', () => {
    shallow(<MyMainApp />)
  });
  it('Ticket render', () => {
    const wrapper = mount(<Ticket name={name} />)
    const button = wrapper.find('button')
    const title = wrapper.find('h2.title').text()
    expect(title).toEqual(name)
    expect(button.length).toEqual(2)
  })
});

describe('snapshots', () => {
  it('MyComponent snapshot', () => {
    const tree = shallow(<MyComponent />)
    expect(toJson(tree)).toMatchSnapshot()
  });

  it('Footer snapshot', () => {
    const tree = shallow(<Footer />)
    expect(toJson(tree)).toMatchSnapshot()
  });

  it('MyMainApp snapshots', () => {
    const tree = renderer.create(<MyMainApp />).toJSON()
    expect(tree).toMatchSnapshot()
  });

  it('Ticket snapshot', () => {
    const wrapper = shallow(<Ticket name={name} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})