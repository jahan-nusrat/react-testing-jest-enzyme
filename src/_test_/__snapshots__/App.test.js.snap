// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Footer snapshot 1`] = `
<div>
  <hr />
  <span
    className="phone"
  >
    Customer Service: 1-889-34343-43
  </span>
</div>
`;

exports[`snapshots MyComponent snapshot 1`] = `
<div
  style={
    Object {
      "btn": Object {
        "backgroundColor": "lightgreen",
        "border": "1px solid yellow",
        "borderRadius": "5px",
        "cursor": "pointer",
        "fontSize": "15px",
        "outline": "none",
        "padding": "0.5rem 1rem",
      },
      "fontFamily": "monospace",
      "margin": "auto",
      "width": "500px",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "lightgreen",
        "border": "1px solid yellow",
        "borderRadius": "5px",
        "cursor": "pointer",
        "fontSize": "15px",
        "outline": "none",
        "padding": "0.5rem 1rem",
      }
    }
  >
    Toggle
  </button>
  <p>
    Lorem ipsum dolor sit
  </p>
</div>
`;

exports[`snapshots MyMainApp snapshots 1`] = `
<div
  className="app container"
>
  <div
    className="section"
  >
    <h1
      className="has-text-centered title is-1"
    >
      Welcome in the personal finance app!
    </h1>
    <div
      className="columns"
    >
      <div
        className="column"
      >
        <div
          className="box"
        >
          <h4
            className="title is-4"
          >
            Your account balance:
          </h4>
          <div
            className="amount balance"
          >
            $
          </div>
          <button
            className="button is-info"
            id="balance-button"
            onClick={[Function]}
          >
            Send 100$
          </button>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="box"
        >
          <h4
            className="title is-4"
          >
            Your savings balance:
          </h4>
          <div
            className="amount savings"
          >
            $
          </div>
          <button
            className="button is-info"
            onClick={[Function]}
          >
            Send 100$
          </button>
        </div>
      </div>
      <div
        className="notification is-danger hidden"
      >
        <button
          className="delete"
          onClick={[Function]}
        />
        <p>
          Your account balance is very low.
        </p>
        <p>
          You can't transfer more money today.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`snapshots Ticket snapshot 1`] = `
<div
  style={
    Object {
      "margin": "auto",
      "textAlign": "center",
      "width": "300px",
    }
  }
>
  <h2
    className="title"
  >
    Hello world
  </h2>
  <button
    className="increment"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "yellow",
        "cursor": "pointer",
        "padding": "0.5rem 1rem",
      }
    }
  >
    Add
  </button>
  <button
    className="decrement"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "yellow",
        "cursor": "pointer",
        "padding": "0.5rem 1rem",
      }
    }
  >
    Reduce
  </button>
  <h2
    className="total"
  >
    Count: 
    0
  </h2>
</div>
`;
